lib/services/publish-service.ts

```diff
@@ -0,0 +1,211 @@
+            }
+
+            if (!ftpConnectionData.Password) {
+                ftpConnectionData.Password = this.$prompter.getPassword("Password:", {allowEmpty: true}).wait();
+            }
+
+            var id = parseInt(idOrUrl);
+
+            // the check wether it contains a dot is performed
+            // in case the user tries to publish to an IP address
+            if (!isNaN(id) && idOrUrl.indexOf('.') < 0) {
+                this.validateId(id);
+
+                // (id - 1) is the index of the object to be deleted
+                ftpConnectionData.RemoteUrl = this.ftpPublishConnections[id - 1].publishUrl;
+            }
+
+            var projectDir = this.$project.getProjectDir().wait();
+            this.$projectFilesManager.excludeFile(projectDir, PublishService.JSON_PUBLISH_FILE_NAME);
+            this.$project.importProject().wait();
+            this.$logger.printInfoMessageOnSameLine("Publishing");
+
+            try {
+                this.$progressIndicator.showProgressIndicator(this.$server.publish.publishToFtp(projectName, projectName, ftpConnectionData), 6000).wait();
+            } catch (err) {
+                this.$logger.printInfoMessageOnSameLine(os.EOL);
+                if (err.message === PublishService.ERROR_MESSAGE_WHEN_PROJECT_EXISTS_ON_REMOTE) {
+                    this.$errors.failWithoutHelp("This project already exists on the remote server. " + os.EOL + 
+                                    "To remove all files and upload the new ones on the remote run --force option");
+                }
+
+                throw err;
+            }
+
+            this.$logger.printInfoMessageOnSameLine(os.EOL);
+            this.$logger.info("Project '%s' published successfully.", projectName);
+        }).future<void>()();
+    }
+
+    public listAllConnections(): IFuture<void> {
+        return (() => {
+            this.listConnections(this.ftpPublishConnections);
+
+            if (!this.ftpPublishConnections.length) {
+                this.$logger.info("No connections found. To add a connection, run `connection add`");
+            }
+        }).future<void>()();
+    }
+
+    public addConnection(name: string, publishUrl: string): IFuture<void> {
+        return (() => {
+            var ftpPublishConnection: IPublishConnection = {
+                type: PublishService.ALLOWED_COONECTION_TYPE,
+                publicUrl: PublishService.PUBLIC_URL_DEFAULT_VALUE,
+                publishUrl: publishUrl,
+                name: name
+            };
+
+            if (!ftpPublishConnection.name) {
+                ftpPublishConnection.name = this.$prompter.getString("Name:").wait();
+            }
+
+            if (!ftpPublishConnection.publishUrl) {
+                ftpPublishConnection.publishUrl = this.$prompter.getString("Publish URL:").wait();
+            }
+
+            this.allPublishConnections.push(ftpPublishConnection);
```

Comment:
Actually it was strange and confusing to me but after talking with @mitko in person we decided that we need to support this case.  Actually we may have two connnections with the same name and url but with different site url. So I think it is better to show a warning.